OSDOCS-17043_d#AWS and Azure CQAs (#111453)
openshift/openshift-docs: modules/machineset-creating-azure-ultra-disk.adoc

get secret <role>-user-data \ <1>
--template='{{index .data.userData | base64decode}}' | jq > userData.txt <2>
----
+
where:
+
--
`<role>`:: Replace with `{machine-role}`.

> `userData.txt`:: Specify `userData.txt` as the name of the new custom secret.

Reviewer: ```suggestion
`userData.txt`:: Specifies `userData.txt` as the name of the new custom secret.
```

            managedDisk:
              storageAccountType: UltraSSD_LRS
          userDataSecret:
            name: <role>-user-data-x5
endif::mapi[]
----
+
where:
+
--

> `spec.template.spec.metadata.labels.disk`:: Specifies a label to use to select a node that is created by this machine set. This procedure uses `disk.ultrassd` for this value.

Reviewer: ```suggestion
`spec.template.spec.metadata.labels.disk`:: Specifies a label to use to select a node that is created by this machine set. The example uses `disk.ultrassd` for this value.
```

            managedDisk:
              storageAccountType: UltraSSD_LRS
          userDataSecret:
            name: <role>-user-data-x5
----
+
where:
+
--

> `spec.template.spec.metadata.labels.disk`:: Specifies a label to use to select a node that is created by this machine set. This procedure uses `disk.ultrassd` for this value.

Reviewer: ```suggestion
`spec.template.spec.metadata.labels.disk`:: Specifies a label to use to select a node that is created by this machine set. The example uses `disk.ultrassd` for this value.
```

apiVersion: storage.k8s.io/v1
kind: StorageClass
metadata:
  name: ultra-disk-sc
parameters:
  cachingMode: None
  diskIopsReadWrite: "2000"
  diskMbpsReadWrite: "320"
  kind: managed
  skuname: UltraSSD_LRS
provisioner: disk.csi.azure.com
reclaimPolicy: Delete
volumeBindingMode: WaitForFirstConsumer
----
+
where:
+
--

> `metadata.name`:: Specifies the name of the storage class. This procedure uses `ultra-disk-sc` for this value.

Reviewer: ```suggestion
`metadata.name`:: Specifies the name of the storage class. The example uses `ultra-disk-sc` for this value.
```
DO same for any remaining items.

> `parameters.diskIopsReadWrite`:: Specifies the number of IOPS for the storage class.

Reviewer: Expand IOPS?

`parameters.diskMbpsReadWrite`:: Specifies the throughput in MBps for the storage class.
`provisioner`:: For {azure-full} Kubernetes Service (AKS) version 1.21 or later, use `disk.csi.azure.com`. For earlier versions of AKS, use `kubernetes.io/azure-disk`.

> `volumeBindingMode`:: Optional: Specifies this parameter to wait for the creation of the pod that will use the disk.

Reviewer: ```suggestion
`volumeBindingMode`:: Optional parameter. Specifies this parameter to wait for the creation of the pod that will use the disk.
```
